src/libraries/System.Private.CoreLib/src/System/CoreLib.cs

```diff
@@ -2,11 +2,53 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+
 namespace System
 {
-    // This class is used to define the name of the base class library
+    /// <summary>
+    /// Defines capabilities that cut across the whole CoreLib library.
+    /// </summary>
     internal static class CoreLib
     {
+        /// <summary>
+        /// Defines the name of the base class library
+        /// </summary>
         public const string Name = "System.Private.CoreLib";
+
+        /// <summary>
+        /// InitializeILLinkTrimmableCode is a method that is never called, but exists
+        /// in order to define feature switches for the ILLinker to analyze.
+        /// An application can set these feature switches to tell the linker to trim code
+        /// that is not needed by the application.
+        /// </summary>
+        private static void InitializeILLinkTrimmableCode()
+        {
+            if (!DebugMode.None)
```

Comment:
`!Debug.Mode.None` is hard to read double-negation.

Would it be better to attach it to the existing `Debugger` type and call it e.g. `Debugger.IsSupported` ?

Reply:
Should I also change the name of the AppContext string to match the code type and property name?

One concern I was trying to keep was to make the default value ‘false’ (what the logical value is when the feature switch wasn’t specified).

Reply:
I think having a switch with default value `true` is fine. I do not think we need to inventing unnatural names for switches to make their default value `false`. The defaults can be different for different configurations or over time. A good example is `TieredCompilation`. It was off by default in 2.1 and it is on by default in 3.0+.

Reply:
`Debugger.IsSupported` sounds better but is it "fine" to use it to block debug symbols support across the whole framework (just raising that for clarity) ?

Reply:
I've renamed the property to `Debugger.IsSupported`. Let me know what you think.

Reply:
`Debugger.IsSupported` looks good to me.

I am not sure whether it makes sense to have one property to disable both debugger support and line numbers in stacktraces; or whether the line numbers in stacktraces should be a different property. Marek raised this as well.

Reply:
I don't think this feature will be used unless we set it as default where appropriate and disabling by default the line numbers in stack-traces could be too much for most developers. Thinking about the name again `Debugger.IsSupported` can be confusing if we want to make it the default for some Release configurations and perhaps `Debugger.IsEnabled` which you override with `true` in release configs if needed could work better.